sources/network/windows/io_service.cpp

```diff
@@ -247,14 +307,19 @@ io_service::track(const tcp_socket& socket, const event_callback_t& rd_callback,
   track_info.wr_callback        = wr_callback;
   track_info.marked_for_untrack = false;
 
+  //Make sure we clear these because we might be in a read callback when we
+  //detect the socket failure and reconnect with reused socket handle
+  track_info.is_executing_rd_callback = false;
+  track_info.is_executing_wr_callback = false;
```

Comment:
I think I have to double check that.
For some reason, I think I removed these lines in a past fix, probably for concurrency issues?
Need to double check if that's true and if so, why? Then, will need to understand why it is back here to possibly find another way to do it.

Reply:
my bad, seems like those lines were never there in the past.
Need to investigate for possible concurrency issue. Basically this can potentially cause some troubles in applications for which order is important: this might lead to multiple callback running at once for the same socket, causing data to be unordered.
For HTTP server applications, it can also lead to response order be different from request order, so I think this change can't stay.
Need to understand the reason and find an alternative.

Reply:
though, should be fine if only in track?
Really need to investigate the consequences.